static/ankersrv.js

```diff
@@ -50,12 +50,107 @@ $(function () {
         }, +alert_timeout);
     });
 
+    /**
+     * Get temperature from input
+     * @param {number} temp Temperature in Celsius
+     * @returns {number} Rounded temperature
+     */
+    function getTemp(temp) {
+        return Math.round(temp / 100);
+    }
+
+    /**
+     * Calculate the percentage between two numbers
+     * @param {number} layer
+     * @param {number} total
+     * @returns {number} percentage
+     */
+    function getPercentage(progress) {
+        return Math.round(((progress / 100) * 100) / 100);
+    }
+
+    /**
+     * Truncate a string to a specific length
+     * @param {string} str
+     * @param {number} maxLength
+     * @returns {string} Truncated string
+     */
+    function truncStr(str, maxLength) {
+        if (str.length > maxLength) {
+            return str.slice(0, maxLength) + "...";
+        } else {
+            return str;
+        }
+    }
+
+    /**
+     * Convert time in seconds to hours, minutes, and seconds format
+     * @param {number} seconds
+     * @returns {string} Formatted time string
+     */
+    function getTime(seconds) {
+        const hours = Math.floor(seconds / 3600);
+        const minutes = Math.floor((seconds % 3600) / 60);
+        const remainingSeconds = seconds % 60;
+
+        const timeString = `${hours.toString().padStart(2, "0")}:${minutes
+            .toString()
+            .padStart(2, "0")}:${remainingSeconds.toString().padStart(2, "0")}`;
+
+        return timeString;
+    }
+
+    /**
+     * Get the filament weight based on its length
+     * @param {number} length
+     * @returns {number} Weight
+     */
+    function getWeight(length) {
+        const diameter = 1.75;
+        const density = 1.24;
+        const adjustedLength = length / 100;
+        const volume = Math.PI * (diameter / 2) ** 2 * adjustedLength;
+        const weight = (volume * density).toFixed(2);
+        return weight;
+    }
+
     /**
      * Opens a websocket connection and outputs any incoming message data to console
      */
     socket = new WebSocket("ws://" + location.host + "/ws/mqtt");
     socket.addEventListener("message", (ev) => {
-        console.log(JSON.parse(ev.data));
+        const data = JSON.parse(ev.data);
+        if (data.commandType == 1001) {
```

Comment:
Since we have `.stl` files for the protocol specification, we can easily generate some a `.js` file with symbolic names for these MQTT constants.

This would make it a lot easier to follow what the decode logic is here.

I've pushed a new branch that demonstrates this. (https://github.com/Ankermgmt/ankermake-m5-protocol/blob/chrivers/basic-libflagship-js-support/static/libflagship.js)

If that works, I'll get it merged into the upcoming branch.